src/runtime/create-runtime-stories.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-env browser */
+import { logger } from '@storybook/client-logger';
+import type { Meta, StoryObj } from '@storybook/svelte';
+import { mount, unmount, type ComponentType } from 'svelte';
+
+import StoriesExtractor from './StoriesExtractor.svelte';
+import StoryRenderer from './StoryRenderer.svelte';
+import type { StoriesRepository } from './contexts/extractor.svelte.js';
+
+const createFragment = document.createDocumentFragment
+  ? () => document.createDocumentFragment()
+  : () => document.createElement('div');
+
+/**
+ * @module
+ * Called from a bundler.
+ *
+ * It mounts the Stories components in a context which disables
+ * the rendering of every `<Story />`,
+ * but instead collects names and properties.
+ *
+ * For every discovered `<Story />`, it creates a `StoryFn` which
+ * instantiate the main Stories component: Every Story but
+ * the one selected is disabled.
+ */
+// TODO: I'm not sure the 'meta' is necessary here. As long as it's default exported, SB should internally combine it with the stories. Except for the play logic below, that looks funky, need to ask Pablo about that.
+export const createRuntimeStories = <TMeta extends Meta>(Stories: ComponentType, meta: TMeta, codeByStoryMap: Record<string, string>) => {
```

Comment:
I think the second argument `codeByStoryMap` might not be necessary.

Reference: https://github.com/storybookjs/addon-svelte-csf/pull/181#issuecomment-2143539873

The way how I solved **step 2** is:

I update every compiled `Story` object expression responsible for passing props of the `<Story />` svelte component.
That means, every `Story` will have automatically filled prop `parameters.docs.source.code` with a string slice from the original code.

So, what you require is already accessible at runtime. I'll create another comment below, where we can access the code slice from **step 2**.